src/library/scala/collection/mutable/ArrayBuilder.scala

```diff
@@ -634,65 +726,37 @@ object ArrayBuilder {
     override def toString = "ArrayBuilder.ofBoolean"
   }
 
-  /** A class for array builders for arrays of `Unit` type. */
+  /** A class for array builders for arrays of `Unit` type. It can be reused. */
   @deprecatedInheritance("ArrayBuilder.ofUnit is an internal implementation not intended for subclassing.", "2.11.0")
   class ofUnit extends ArrayBuilder[Unit] {
 
-    private var elems: Array[Unit] = _
-    private var capacity: Int = 0
     private var size: Int = 0
 
-    private def mkArray(size: Int): Array[Unit] = {
-      val newelems = new Array[Unit](size)
-      if (this.size > 0) Array.copy(elems, 0, newelems, 0, this.size)
-      newelems
-    }
-
-    private def resize(size: Int) {
-      elems = mkArray(size)
-      capacity = size
-    }
-
-    override def sizeHint(size: Int) {
-      if (capacity < size) resize(size)
-    }
-
-    private def ensureSize(size: Int) {
-      if (capacity < size || capacity == 0) {
-        var newsize = if (capacity == 0) 16 else capacity * 2
-        while (newsize < size) newsize *= 2
-        resize(newsize)
-      }
-    }
-
     def +=(elem: Unit): this.type = {
-      ensureSize(size + 1)
-      elems(size) = elem
       size += 1
       this
     }
 
-    override def ++=(xs: TraversableOnce[Unit]): this.type = xs match {
-      case xs: WrappedArray.ofUnit =>
-        ensureSize(this.size + xs.length)
-        Array.copy(xs.array, 0, elems, this.size, xs.length)
-        size += xs.length
-        this
-      case _ =>
-        super.++=(xs)
+    override def ++=(xs: TraversableOnce[Unit]): this.type = {
+      size += xs.size
+      this
     }
 
-    def clear() {
-      size = 0
-    }
+    /** Discards any existing elements.
+      *
+      * If `result` was called previously, this will reset the builder to build a new instance.
+      */
+    def clear() { size = 0 }
 
-    def result() = {
-      if (capacity != 0 && capacity == size) elems
-      else mkArray(size)
-    }
+    /** Returns the newly created Array.
+      *
+      * After calling `result`, the result of all methods of this builder are undefined until `clear` is called.
+      * Following a `clear`, the builder will be ready to build an additional instance of the same kind of Array.
+      */
+    def result() = new Array[Unit](size)
```

Comment:
Technically, this changes the behavior of this builder. It will return an array filled with `null`s, whereas the previous implementation would return an array filled with `BoxedUnit.UNIT`.